go/vt/vtgate/endtoend/vschema_test.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2019 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endtoend
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"testing"
+	"time"
+
+	"vitess.io/vitess/go/mysql"
+)
+
+func TestVSchema(t *testing.T) {
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	//hardcoded cell name from vtcombo cluster.
+	cell := "test"
+
+	//Fetch VSchema and verify if it has a Keyspace
+	vSchema, _ := cluster.GetVSchema(cell)
+	if got, want := hasValidKeyspace(vSchema), true; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+
+	//Delete the current VSchema, it should return an error if we try to refer it.
+	cluster.DeleteVSchema(cell)
+	_, err = cluster.GetVSchema(cell)
+	want := "node doesn't exist"
+	if err == nil || !strings.Contains(err.Error(), want) {
+		t.Errorf("Deleted schema: %v, must contain %s", err, want)
+	}
+
+	//Rebuild the VSchema in the cluster
+	cluster.RebuildVSchema(cell)
+	time.Sleep(5 * time.Second)
```

Comment:
Why do we need to wait 5 seconds here? It's better to check for the condition in a loop and sleep for 0.1 seconds and timeout after 5 seconds (or 10 seconds).
Here is an example:
https://github.com/vitessio/vitess/blob/reparent-refactor/go/vt/wrangler/testlib/external_reparent_test.go#L103

Reply:
It turns out we don't need timeout at all, I copied it from Python but not sure if it is needed. 
I executed this test many time and it runs fine.  If you still feel we need a loop and timeout then I will add it. 

As far as I understand this test will just update topology server via vtctlclient. So vtGate should not be involved. Let me know if I am missing something here.

Reply:
Since you are using `RunCommandAndWait`, this should be fine.